src/seedu/addressbook/commands/employee/EmployeeEditCommand.java

```diff
@@ -0,0 +1,204 @@
+            + "[pos/POSITION]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + "p/97654321 ";
+
+    public static final String MESSAGE_EDIT_EMPLOYEE_SUCCESS = "Edited Person: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_EMPLOYEE = "This person already exists in the address book.";
+
+    private final EditEmployeeDescriptor editEmployeeDescriptor;
+
+    /**
+     * @param targetVisibleIndex of the person in the filtered person list to edit
+     * @value editEmployeeDescriptor details to edit the employee with
+     */
+    public EmployeeEditCommand(int targetVisibleIndex,
+                               String phone,
+                               String email,
+                               String address,
+                               String position) throws IllegalValueException{
+        super(targetVisibleIndex);
+        this.editEmployeeDescriptor = new EditEmployeeDescriptor(phone, email, address, position);
+    }
+
+    // check for any non null parameters should already be done, probably in parser
+    @Override
+    public CommandResult execute() {
+        try {
+            final ReadOnlyEmployee employeeToEdit = getTargetEmployee();
+            Employee editedEmployee = createEditedEmployee(employeeToEdit, editEmployeeDescriptor);
+
+            rms.editEmployee(employeeToEdit, editedEmployee);
+
+            return new CommandResult(String.format(MESSAGE_EDIT_EMPLOYEE_SUCCESS, editedEmployee));
+
+        } catch (IndexOutOfBoundsException ie) {
+            return new CommandResult(Messages.MESSAGE_INVALID_EMPLOYEE_DISPLAYED_INDEX);
+        } catch (EmployeeNotFoundException enfe) {
+            return new CommandResult(Messages.MESSAGE_EMPLOYEE_NOT_IN_RMS);
+        }
+
+    }
+
+    /**
+     * Creates and returns an {@code Employee} with the details of {@code employeeToEdit}
+     * edited with {@code editEmployeeDescriptor}.
+     */
+    private static Employee createEditedEmployee(ReadOnlyEmployee employeeToEdit, EditEmployeeDescriptor editEmployeeDescriptor) {
+
+        EmployeeName updatedName = employeeToEdit.getName();
+        EmployeePhone updatedPhone = checkPhone(editEmployeeDescriptor.getPhone(),employeeToEdit.getPhone());
+        EmployeeEmail updatedEmail = checkEmail(editEmployeeDescriptor.getEmail(), employeeToEdit.getEmail());
+        EmployeeAddress updatedAddress = checkAddress(editEmployeeDescriptor.getAddress(), employeeToEdit.getAddress());
+        EmployeePosition updatedPosition = checkPosition(editEmployeeDescriptor.getPosition(), employeeToEdit.getPosition());
+
+        return new Employee(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedPosition);
+    }
+
+    /**
+    * Check for new phone value.
+    */
+    private static EmployeePhone checkPhone(EmployeePhone newEdit, EmployeePhone oldInfo){
+        if (newEdit.value == "00000000")
```

Comment:
The "empty" values should be defined as static constants in the class responsible for it.
Ex: `public static final String EMPTY_EMPLOYEE_PHONE = "00000000";` in `EmployeePhone.java`

Reply:
Edited.